mm/mm_heap/mm_malloc.c

```diff
@@ -114,11 +122,11 @@ FAR void *mm_malloc(FAR struct mm_heap_s *heap, size_t size)
 
   free_delaylist(heap);
 
-  /* Ignore zero-length allocations */
+  /* Use unique ptr for zero-length allocatoins */
 
   if (size < 1)
     {
-      return NULL;
+      return &g_zero_alloc;
```

Comment:
Global variable should not be used

Reply:
Yes. That is true. But what if we make it a member of `struct mm_heap_s`?

Reply:
Discard what I just wrote above and going with minimum size allocating seems to be a correct path if we decide to return non-NULL pointer.

Reply:
Also the latest version of POSIX has nothing about "unique pointer" and description is changed to `If the size of the space requested is 0, the behavior is implementation-defined: either a null pointer shall be returned, or the behavior shall be as if the size were some non-zero value, except that the behavior is undefined if the returned pointer is used to access an object.`, so treating `malloc(0)` as `malloc(1)` is the only correct way.

Reply:
Done. Removed global vairable.